lib/api/controller/base.js

```diff
@@ -81,6 +81,48 @@ class NativeController extends BaseController {
     return Bluebird.resolve();
   }
 
+  async translateKoncorde (koncordeFilters) {
+    // @todo throw "koncorde_unknown_keyword" when Koncorde will throw
+    // just the keyword name
+    await this.kuzzle.koncorde.validate(koncordeFilters);
+
+    if (typeof koncordeFilters !== 'object') {
+      throw kerror.get(
+        'api',
+        'assert',
+        'invalid_type',
+        'body.query',
+        'object');
+    }
+
+    try {
+      return await this.ask('core:storage:public:translate', koncordeFilters);
+    }
+    catch (error) {
+      throw kerror.get(
+        'api',
+        'assert',
+        'koncorde_restricted_keyword',
+        error.keyword.type,
```

Comment:
This is a weird catch... It's like the catch knows exactly the type of the error that's thrown from the `ask` call. It's some kind of hard coupling, IMHO. We should at least check the type of the error to ensure the subfield `keyword` exists inside it.